cmake/std/trilinosprhelpers/TrilinosPRConfigurationBase.py

```diff
@@ -0,0 +1,587 @@
+#!/usr/bin/env python
+# -*- mode: python; py-indent-offset: 4; py-continuation-offset: 4 -*-
+"""
+This file contains the base class for the Pull Request test driver.
+"""
+import configparser
+import multiprocessing
+import os
+import re
+import subprocess
+import sys
+from textwrap import dedent
+
+sys.dont_write_bytecode = True
+
+
+from . import setenvironment
+from . import sysinfo
+from . import jenkinsenv
+
+
+
+class TrilinosPRConfigurationBase(object):
+    """
+    Trilinos Pull Request configuration driver. This should be
+    treated as an Abstract Base Class because the `execute()`
+    method is implemented as a stub.
+
+    Note: Attributes / Properties prefixed with `arg_` come from the
+        command line arguments that is passed into this class c'tor.
+        We provide these properties to give a read-only access to the
+        properties.
+
+    Attributes:
+        arg_github_pr_number: PR Number on Github
+        arg_job_number: Job Number of the Jenkins job.
+        arg_req_mem_per_core: Memory required per core (GB)
+        arg_max_cores_allowed: Max cores allowed for building (i.e., make -j <num cores>)
+        arg_num_concurrent_tests: Testing concurrency (i.e., ctest -j <num cores>)
+        arg_package_enables_file: Path to the `packageEnables.cmake` file.
+        arg_workspace_dir: Path to the workspace (this would be the ${WORKSPACE}
+            variable set by Jenkins.)
+        arg_pr_config_file: The config.ini file that specifies the configuration to load.
+        arg_pr_jenkins_job_name: The Jenkins Job Name.
+        subprojects_file:
+        working_directory_ctest: Gen. working dir where TFW_testing_single_configure_prototype
+            is executed from.
+        trilinos_pr_env: Environment variable info specific to Trilinos Jobs.
+        config_data:
```

Comment:
Missing descriptions for `subprojects_file` and `config_data`?

Reply:
I was working on this yesterday and had to quit for the day.

It is documented in the `config_data` property.